src/oscar/apps/partner/abstract_models.py

```diff
@@ -1,260 +0,0 @@
-from django.db import models
-from django.utils.encoding import python_2_unicode_compatible
-from django.utils.translation import ugettext_lazy as _, pgettext_lazy
-
-from oscar.core.utils import get_default_currency
-from oscar.core.compat import AUTH_USER_MODEL
-from oscar.models.fields import AutoSlugField
-from oscar.apps.partner.exceptions import InvalidStockAdjustment
-
-
-@python_2_unicode_compatible
-class AbstractPartner(models.Model):
-    """
-    A fulfillment partner. An individual or company who can fulfil products.
-    E.g. for physical goods, somebody with a warehouse and means of delivery.
-
-    Creating one or more instances of the Partner model is a required step in
-    setting up an Oscar deployment. Many Oscar deployments will only have one
-    fulfillment partner.
-    """
-    code = AutoSlugField(_("Code"), max_length=128, unique=True,
-                         populate_from='name')
-    name = models.CharField(
-        pgettext_lazy(u"Partner's name", u"Name"), max_length=128, blank=True)
-
-    #: A partner can have users assigned to it. This is used
-    #: for access modelling in the permission-based dashboard
-    users = models.ManyToManyField(
-        AUTH_USER_MODEL, related_name="partners",
-        blank=True, verbose_name=_("Users"))
-
-    @property
-    def display_name(self):
-        return self.name or self.code
-
-    @property
-    def primary_address(self):
-        """
-        Returns a partners primary address. Usually that will be the
-        headquarters or similar.
-
-        This is a rudimentary implementation that raises an error if there's
-        more than one address. If you actually want to support multiple
-        addresses, you will likely need to extend PartnerAddress to have some
-        field or flag to base your decision on.
-        """
-        addresses = self.addresses.all()
-        if len(addresses) == 0:  # intentionally using len() to save queries
-            return None
-        elif len(addresses) == 1:
-            return addresses[0]
-        else:
-            raise NotImplementedError(
-                "Oscar's default implementation of primary_address only "
-                "supports one PartnerAddress.  You need to override the "
-                "primary_address to look up the right address")
-
-    def get_address_for_stockrecord(self, stockrecord):
-        """
-        Stock might be coming from different warehouses. Overriding this
-        function allows selecting the correct PartnerAddress for the record.
-        That can be useful when determining tax.
-        """
-        return self.primary_address
-
-    class Meta:
-        abstract = True
-        app_label = 'partner'
-        permissions = (('dashboard_access', 'Can access dashboard'), )
```

Comment:
There's one template left in oscar code where this permission is used. I don't know if we use that template.